public/pages/ConfigureForecastModel/components/Settings/Settings.tsx

```diff
@@ -0,0 +1,331 @@
+                          showRange
+                          levels={[
+                            { min: 1, max: 40, color: 'success' },
+                            { min: 40, max: 180, color: 'warning' },
+                          ]}
+                          aria-label="Horizon slider"
+                        />
+                        <EuiSpacer size="xs" />
+                        <EuiText size="xs" color="subdued">
+                          Recommended horizon is up to 40 intervals.
+                        </EuiText>
+                      </EuiPanel>
+                    </div>
+                  </>
+                )}
+              </div>
+            </FormattedFormRow>
+          );
+        }}
+      </Field>
+
+      <EuiSpacer size="l" />
+
+      {/* HISTORY */}
+      <Field name="history" validate={validatePositiveInteger}>
+        {({ field, form }: FieldProps<number>) => {
+          const interval = (form as FormikProps<any>).values.interval;
+          const value = field.value ?? '';
+
+          function getHistoryHelpText(history: number, intv: number) {
+            if (!intv || !history) return 'Minimum history: 40 intervals.';
+            const totalMinutes = history * intv;
+            const hours = Math.floor(totalMinutes / 60);
+            const mins = totalMinutes % 60;
+            return `${history} intervals = ${hours} hour${
+              hours === 1 ? '' : 's'
+            }${
+              mins ? ` ${mins} minute${mins === 1 ? '' : 's'}` : ''
+            } if the forecasting interval is ${intv} minutes. Minimum history: 40 intervals.`;
+          }
+
+          return (
+            <FormattedFormRow
+              fullWidth
+              title="History"
+              hint={['How far back the model looks for training data.']}
+              hintLink={FORECASTER_DOCS_LINK}
+              isInvalid={isInvalid(field.name, form)}
+              error={getError(field.name, form)}
+              helpText={getHistoryHelpText(field.value, interval)}
+            >
+              {/* gutterSize="none" ensures there's no default spacing between flex items */}
+              <EuiFlexGroup
+                gutterSize="none"
+                alignItems="center"
+                style={{ maxWidth: FIELD_MAX_WIDTH }}
+              >
+                
+                <EuiFlexItem grow={false}>
+                  <div style={{ width: INPUT_SLIDER_WIDTH }}>
+                    <EuiCompressedFieldNumber
+                      min={1}
+                      max={10000}
+                      name={field.name}
+                      id={field.name}
+                      value={value}
+                      disabled={!isEditable}
+                    style={{ width: '100%' }}
+                    onChange={(e) => {
+                      const raw = e.target.value;
+                      form.setFieldValue(field.name, raw === '' ? '' : Number(raw));
+                    }}
```

Comment:
Same as in the AdvancedSetting file above, the onChange method here is identical and repeated multiple times